graspologic/plot/plot_matrix.py

```diff
@@ -0,0 +1,732 @@
+
+    return meta, group_class, class_order, item_order, color_class
+
+
+def _get_separator_info(meta, group_class):
+    if meta is None or group_class is None:
+        return None
+
+    front = meta.groupby(by=group_class, sort=False).first()
+    front_sep_inds = front["sort_idx"].values
+    end = meta.groupby(by=group_class, sort=False).last()
+    back_sep_inds = end["sort_idx"].values
+
+    return front_sep_inds, back_sep_inds
+
+
+def _item_to_df(item, name, length):
+    if item is None:
+        return None, []
+
+    if isinstance(item, pd.Series):
+        _check_length(item, name, length)
+        item_meta = item.to_frame(name=f"{name}_0")
+    elif isinstance(item, list):
+        if len(item) == length:  # assuming elements of list are metadata
+            item = [item]
+        for elem in item:
+            _check_length(elem, name, length)
+        item_meta = pd.DataFrame({f"{name}_{i}": elem for i, elem in enumerate(item)})
+    elif isinstance(item, np.ndarray):
+        if item.ndim > 2:
+            raise ValueError(f"Numpy array passed as {name} must be 1 or 2d.")
+        _check_length(item, name, length)
+        if item.ndim < 2:
+            item = np.atleast_2d(item).T
+        item_meta = pd.DataFrame(
+            data=item, columns=[f"{name}_{i}" for i in range(item.shape[1])]
+        )
+    else:
+        raise ValueError(f"{name} must be a pd.Series, np.array, or list.")
+
+    item = list(item_meta.columns.values)
+
+    return item_meta, item
+
+
+def _remove_shared_ax(ax):
+    """
+    Remove ax from its sharex and sharey
+    """
+    # Remove ax from the Grouper object
+    shax = ax.get_shared_x_axes()
+    shay = ax.get_shared_y_axes()
+    shax.remove(ax)
+    shay.remove(ax)
+
+    # Set a new ticker with the respective new locator and formatter
+    for axis in [ax.xaxis, ax.yaxis]:
+        ticker = mpl.axis.Ticker()
+        axis.major = ticker
+        axis.minor = ticker
+        # No ticks and no labels
+        loc = mpl.ticker.NullLocator()
+        fmt = mpl.ticker.NullFormatter()
+        axis.set_major_locator(loc)
+        axis.set_major_formatter(fmt)
+        axis.set_minor_locator(loc)
+        axis.set_minor_formatter(fmt)
+
+
+def draw_colors(
+    ax, ax_type="x", meta=None, divider=None, color_class=None, palette="tab10"
+):
+    """
+    Draw colormap onto the axis to separate the data
+
+    Parameters
+    ----------
+    ax : matplotlib axes object
+        Axes in which to draw the colormap
+    ax_type : char, optional
+        Setting either the x or y axis, by default "x"
+    meta : pd.DataFrame, pd.Series, list of pd.Series or np.array, optional
+        Metadata of the matrix such as class, cell type, etc., by default None
+    divider : AxesLocator, optional
+        Divider used to add new axes to the plot
+    color_class : dict, optional
+        Attribute in meta by which to draw colorbars, by default None
+    palette : str, optional
```

Comment:
can also be a dict?

Reply:
I changed the format description to string or list of strings for color_class and group_class, I don't know if that's accurate

Reply:
can palette be a dict? I think we want this capability

Reply:
Yes, it turns out that palette can be a dict, I just forgot to include it in the documentation